packages/core/src/StateNode.ts

```diff
@@ -673,21 +673,19 @@ class StateNode<
     }
 
     const subStateKeys = keys(stateValue);
-    const subStateNodes: Array<
-      StateNode<TContext, any, TEvent, TTypestate>
-    > = subStateKeys.map((subStateKey) => this.getStateNode(subStateKey));
```

Comment:
Similarly here - the logic that follows this is already pushing `this.getStateNode(subStateKey)` to the `subStateNodes` so this line here could be dropped to avoid duplicate entries being pushed to the final `subStateNodes`